public/lib/downloadHttpConcurrentProgress.js

```diff
@@ -126,18 +129,26 @@
         }
         this._collectMovingAverages = false;
         //pushing results to an array
-        this._results.push(result);
+        this._results.push(result.bandwidth);
         //cancel remaining tests
-        for (var i = 0; i < this._activeTests.length; i++) {
-            if (typeof(this._activeTests[i]) !== 'undefined') {
-                this._activeTests[i].xhr._request.abort();
-            }
-        }
+        this.abortAll();
         //reset Active Tests array
         this._activeTests.length =0;
         //checking if we can continue with the test
         if ((Date.now() - this._beginTime) < this.testLength) {
-            this.start();
+          if(this.isProbing) {
+            this.abortAll();
+            this.probeTotalTime = this.probeTotalTime + result.time;
+            this.probeTotalBytes = this.probeTotalBytes + result.loaded;
+            this.size = (this.probeTimeTimeout-result.time) * result.loaded/result.time;
```

Comment:
Due not lower size during actual testing check new size after group run is not lower